packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -160,11 +161,27 @@ class AOTSnapshotter {
       genSnapshotArgs.add('--no-use-integer-division');
     }
 
+    // The name of the debug file must contain additonal information about
+    // the architecture, since a single build command may produce
+    // multiple debug files.
+    final String archName = darwinArch == null || platform == TargetPlatform.darwin_x64
+      ? getNameForTargetPlatform(platform)
+      : 'ios-${getNameForDarwinArch(darwinArch)}';
+    final String debugFilename = 'app.$archName.debug';
```

Comment:
I like symbols as an extension better, since it will match the symbolize command.

Beyond the entrypoint, there is the current dart+flutter version, pubspec version, in the future dart-defines. Rather than trying to capture all of it, I think we should document and explain that each file is 1-1 with a particular release.